eng/source-build-release-notes.md

```diff
@@ -0,0 +1,24 @@
+You can build $RELEASE_NAME from the repository by cloning the release tag `$TAG` and following the build instructions in the [main README.md](https://github.com/dotnet/dotnet/blob/$TAG/README.md#building).
+
+```sh
+git clone --branch $TAG --depth 1 https://github.com/dotnet/dotnet
+(cd dotnet && ./build.sh)
+```
+
+Alternatively, you can build from the sources attached to this release. Please note that you'll need to provide the release information to the build script. This can be done directly:
+```sh
+curl -Lo dotnet-sdk-source-$TAG.tar.gz https://github.com/dotnet/dotnet/archive/refs/tags/$TAG.tar.gz
+tar zxf dotnet-sdk-source-$TAG.tar.gz
+(cd dotnet-* && ./build.sh --source-repository https://github.com/dotnet/dotnet --source-version $SOURCE_VERSION)
```

Comment:
Here I had to use `dotnet-*` because for some reason I couldn't find the folder inside the tarball, the name doesn't start with the `v`. I'd expect it to be the same as the tag but it's not. We don't have this string anywhere. I hope it's fine?

Reply:
Fixed by not producing the tarball